### PR TITLE
ROX-26475: Compliance Refactor

### PR DESCRIPTION
**Pull request by Maddosaurus**
### Description

Compliance has been steadily growing and found itself with a deep and not easily extendable structure.
This PR restructures Compliance, mostly by moving and reorganizing packages under `/compliance`.
It also updates the cursory references (e.g. tools, Makefile), and renames a function to better reflect its more generalist usage.

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [ ] CI results are inspected

#### Automated testing

No changes to current tests

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [ ] modified existing tests

#### How I validated my change

Automated tests, manual testing by deploying ACS and verifying that legacy compliance and node scanning / indexing still work.

---
This is a before and after overview of the folder structure of compliance:

before

after

### CODE REVIEW

**Review by lvalerom — COMMENTED** [review]
I left a few minor comments. Overall it looks good but there are a couple of things that topped in my head when I saw the reorganization of the code:

1) Would it make sense to extract the `compliance_checks` from the `compliance` package? I know it kind of make sense to keep them there since they are technically the compliance standards part, but at the same time compliance as the structure [Compliance](https://github.com/stackrox/stackrox/blob/fed78a8ca3e2d304988be39f36ee3d54703c36d0/compliance/collection/compliance/compliance.go#L32C1-L39C1) encapsulates the rest of the components (auditlog, node indexing, etc). So to me they look at little bit out of place in that package.

2) Why is the `node` package outside of the `collection` folder? To me it sounds like it could be inside that folder but there might be a reason I don't know.

3) Following up on (2) if there's a reason for `node` to be outside of `collection` I wonder if there might be other packages that could be outside too.

File: compliance/collection/compliance/compliance_checks.go [hunk 1]
```diff
@@ -20,10 +22,10 @@ import (
 
 var log = logging.LoggerForModule()
 
-func runChecks(client sensor.ComplianceService_CommunicateClient,
+func RunChecks(client sensor.ComplianceService_CommunicateClient,
```
[the remark this answers is not in the document]
Comment (Maddosaurus) [reply compliance/collection/compliance/compliance_checks.go:25]: Thanks for flagging this!
I've moved files iteratively until I was happy with the result. At one point I tried to move out `compliance.go`. Looks like I didn't revert that completely.
This also explains some of the weird import aliases now that I think about it :smile:

**Comment by Maddosaurus** [comment]
@lvalerom 
>     1. Would it make sense to extract the `compliance_checks` from the `compliance` package?
We can certainly extract `compliance.go` from this package and move it directly to `/compliance` or `/compliance/collection`. I've also experimented with merging `main.go` with it, but that kills some tests that require parts of this compliance structure to be imported - and you cannot import from a `main` package. Do you have a preference where to place `compliance.go`?
>     2. Why is the `node` package outside of the `collection` folder? To me it sounds like it could be inside that folder but there might be a reason I don't know.
The `node` package contains the two implementations for node scanning. As these are _technically_ not part of Compliance (from a user/UI perspective, e.g. triggering a Compliance run), I decided to move them into their own pacakge.
>     3. Following up on (2) if there's a reason for `node` to be outside of `collection` I wonder if there might be other packages that could be outside too.
With the stance I took in 2, I don't think there are other packages that would need to live outside of collection - maybe except for `compliance.go` (see 1).
The module is right now split into two responsibilities: `/collection`, which is concerned with collecting information for a compliance scan, and `/node`, which is concerned with Node Scanning/Indexing.

File: compliance/collection/compliance_checks/compliance_checks.go [hunk 2]
```diff
@@ -20,10 +22,10 @@ import (
 
 var log = logging.LoggerForModule()
 
-func runChecks(client sensor.ComplianceService_CommunicateClient,
+func RunChecks(client sensor.ComplianceService_CommunicateClient,
```
Comment (lvalerom) [inline compliance/collection/compliance_checks/compliance_checks.go:25]: I think we should add a comment here since this is now public.

**Review by lvalerom — APPROVED** [review]
LGTM. Just two minor things:

- I think we should update the images in the PR description with the new folder structure.
- Also I think we should add a comment in the `RunChecks` function.

**Pull request merged**